data-access-layer/bosh/BoshDirectorClient.js

```diff
@@ -33,9 +36,56 @@ class BoshDirectorClient extends HttpClient {
     this.deploymentIpsCache = {};
     this.cacheLoadInProgressForDeployment = {};
     this.cacheLoadInProgress = false;
+    this.uaaObjects = {};
+    //populate UAA objects if uaa based auth is being used by any of the directors
+    this.determineDirectorAuthenticationMethod();
     this.ready = this.populateConfigCache();
   }
 
+  determineDirectorAuthenticationMethod() {
+    _.forEach(config.directors, (directorConfig) => {
+      if (_.get(directorConfig, 'uaa.uaa_auth', undefined)) {
+        let directorName = _.get(directorConfig, 'name', undefined);
+        logger.info('Director ', directorName, ' uses UAA based authentication. Populating UAA objects in directorConfig.');
+        this.populateUAAObjects(directorConfig);
+        if (_.get(this.uaaObjects, directorName + '.uaaClient', undefined) && _.get(this.uaaObjects, directorName + '.tokenIssuer', undefined)) {
+          _.set(directorConfig, 'uaaEnabled', true);
+        } else {
+          //TODO: Fatal error condition. Logging and exiting.
+          logger.error('UAA objects were not popoulated successfully for bosh ', directorName, '.');
```

Comment:
*populated* - spelling mistake
Also, can you club the two error loggers in one?